pallets/asset/src/identifier.rs

```diff
@@ -0,0 +1,176 @@
+use codec::{Decode, Encode};
+use core::convert::TryInto;
+use sp_std::vec::Vec;
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq)]
+pub enum Identifier {
+    CUSIP([u8; 9]),
+    CINS([u8; 9]),
+    ISIN([u8; 12]),
+    LEI([u8; 20]),
+    EMPTY,
+}
+
+impl Default for Identifier {
+    fn default() -> Self {
+        Identifier::EMPTY
+    }
+}
+
+impl Identifier {
+    pub fn cusip(bytes: [u8; 9]) -> Option<Identifier> {
+        (cusip_checksum(&bytes[..8]) == bytes[8] - b'0').then_some(Identifier::CUSIP(bytes))
+    }
+
+    pub fn cins(bytes: [u8; 9]) -> Option<Identifier> {
+        (cusip_checksum(&bytes[..8]) == bytes[8] - b'0').then_some(Identifier::CINS(bytes))
+    }
+
+    pub fn isin(bytes: [u8; 12]) -> Option<Identifier> {
+        let (s1, s2) = bytes
+            .iter()
+            .map(|b| byte_value(*b))
+            .flat_map(|b| if b > 9 { Vec::from([b / 10, b % 10]) } else { Vec::from([b]) })
```

Comment:
Not sure why this was changed to `Vec::from` instead of `vec!`

Reply:
It wouldn’t compile, couldn’t find the vec! macro.

Reply:
Try `use alloc::vec;`